Log exceptions evaluating `IValueProvider` whilst starting a resource

afscrome opened the pull request:
## Description

When an exception happens evaluating `IValueProvider` when starting a resource, log the exception to the console output to provide more clues as to why the resource failed to start.

Makes errors more obvious for #7030 & #6330.

Before:
![image](https://github.com/user-attachments/assets/e903895c-a181-468a-86e2-a8ea7fda4015)

After:
![image](https://github.com/user-attachments/assets/95217437-593b-4a4f-b19f-fe281507b02b)



## Checklist

- Is this feature complete?
  - [X] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [X] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [X] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [X] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [X] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/7032)

afscrome committed:
Log exceptions that occur when evaluating IValueProvider whilst starting a resource.

danmoseley commented:
Thanks! Diagnosability is always good. I'll let code owner sign off.

afscrome commented on src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:
```diff
@@ -1354,8 +1354,7 @@ private async Task CreateExecutableAsync(AppResource er, ILogger resourceLogger,
                 catch (Exception ex)
                 {
-                    resourceLogger.LogCritical("Failed to apply arguments. A dependency may have failed to start.");
-                    _logger.LogDebug(ex, "Failed to apply arguments. A dependency may have failed to start.");
```
I read this as a basic level of detail being logged as Critical, with the full detail only being logged at Debug.  Now the critical level contains the full detail, the debug entry seemed redundant.

The other place in this file that does a `resourceLogger.LogCritical(ex, ...)` follows this same pattern - only logging the exception to the `resourceLogger`, and not `_logger`.

https://github.com/dotnet/aspire/blob/eca04807c4ca03bf44cfec6e13cbafdb092ab92e/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs#L378

I can put the debug logs back if you wat

eerhardt replied:
`resourceLogger` and `_logger` are separate logs / instances. The first is the logs for the resource that shows up in the dashboard. The 2nd is the log for the AppHost itself (since the AppHost uses the ASP.NET/Microsoft.Extensions.Hosting app model).

It isn't redundant because it goes to 2 separate locations.

davidfowl replied:
@afscrome Can you add the logs you removed back?

afscrome replied:
Added debug logs back, although I did add `{ResourceName}` into the `_logger` log entries so you know which resource failed.  (I didn't include this on `resourceLogger` entries since the resource is implied by the `resourceLogger` instance).

afscrome committed:
More IValueProvider failure logging tweaks

- Added missing logging for failures  in container runtime args
- Be more consistent with error message - always including `{ConfigKey}`
- Restore previously removed debug logging, with the addition of `{ResourceName}` to make clear which resource is the problem child.

eerhardt reviewed (approved):
LGTM. Thanks for the contribution!
